### ISSUE TITLE
abstract class Generator should implement GeneratorInterface

### ISSUE THREAD

**Issue by LOBsTerr** [hechoendrupal/drupal-console#3718]
### Issue title

All classes extending Generator, should be updated so we can implement GeneratorInterface

### Problem/Motivation
Code structure improvement

**Comment by LOBsTerr** [comment]
@jmolivas It is the first PR https://github.com/hechoendrupal/drupal-console/pull/3731
I decided to send already right now, because I've already faced some conflicts. Plus it would be easier to review.
Keep in mind that each class Generator implements GeneratorInterface. Later I will move it to abstract class and I will send PR for the core.
Also, in the same time I did refactoring in Generator classes.
